pkg/k8s/api.go

```diff
@@ -112,6 +115,10 @@ func NewK8sAPi(shell shell.Shell, k8sConfigFilesystemPathOverride string, apiHos
 
 	if apiHostAndPortOverride == "" {
```

Comment:
TIOLI Considering how this branching is growing, instead of reusing `apiHostAndPortOverride` to hold something that's not really an override I would consider making the return from the `if`, 
e.g.
```golang
if apiHostAndPortOverride == "" {
	return &kubernetesApi{
		apiSchemeHostAndPort: config.Host,
		config:               config,
	}, nil
} else {
    if !httpPrefix.MatchString(apiHostAndPortOverride) {
	    apiHostAndPortOverride = strings.Join([]string{"https://", apiHostAndPortOverride}, "")
	}
	return &kubernetesApi{
		apiSchemeHostAndPort: apiHostAndPortOverride,
		config:               config,
	}, nil
}
```